### PR TITLE
ipa: retry restore if it fails to bind to LDAP

### PR DESCRIPTION
**Pull request by pbrezina**
ipa-restore sometimes returns 0 but does not restore IPA correctly.

```
+ ipa-restore --unattended --password Secret123 --data --online /tmp/tmp.aBMkEEeqbD/ipa
Preparing restore from /tmp/tmp.aBMkEEeqbD/ipa on master.ipa.test
Performing DATA restore from DATA backup
Temporary setting umask to 022
Each master will individually need to be re-initialized or
re-created from this one. The replication agreements on
masters running IPA 3.1 or earlier will need to be manually
re-enabled. See the man page for details.
Disabling all replication.
Starting Directory Server
Restoring from userRoot in IPA-TEST
Unable to bind to LDAP server: Operations error:
Restoring from ipaca in IPA-TEST
Waiting for LDIF to finish
Restoring umask to 18
The ipa-restore command was successful
```

### CODE REVIEW

**Comment by alexey-tikhonov** [comment]
```
sssd_test_framework/misc/ssh.py:123: error: Missing return statement  [return]
```

**Comment by pbrezina** [comment]
Fixed. By the way, the first commit is actually not needed in the end, but I left it there anyway.

**Comment by alexey-tikhonov** [comment]
Looks like this doesn't help.

https://github.com/SSSD/sssd/pull/7584 was pinned to use github.com/pbrezina/sssd-test-framework@iparestore still it fails the same way - https://github.com/SSSD/sssd/actions/runs/10938362534/job/30373780704?pr=7584

**Comment by pbrezina** [comment]
Ok, I will run it and get more logs. Maybe we should run --mh-collect-artifacts=always or at least --mh-collect-logs=always. I used on-failure only to get fast access to logs of failed tests, but it make it difficult to debug this kind of issues.

**Comment by pbrezina** [comment]
@alexey-tikhonov Look like it works now.

File: sssd_test_framework/hosts/ipa.py [hunk 1]
```diff
@@ -171,29 +171,33 @@ def restore(self, backup_data: Any | None) -> None:
...
+                restore "{backup_path}/krb5.keytab" /etc/krb5.keytab
+                restore "{backup_path}/config" /etc/sssd
+                restore "{backup_path}/logs" /var/log/sssd
+                restore "{backup_path}/lib" /var/lib/sss
+                """,
+                log_level=ProcessLogLevel.Error,
+            )
+
+        backup_path = str(backup_data)
+        self.logger.info(f"Restoring IPA server from {backup_path}")
```
Comment (alexey-tikhonov) [inline sssd_test_framework/hosts/ipa.py:201]: I wonder what changed in those two lines above so that github shows them as -+?

**Review by alexey-tikhonov — APPROVED** [review]

**Comment by alexey-tikhonov** [comment]
> @alexey-tikhonov Look like it works now.

Looks so - got 3 green runs in a row.

File: sssd_test_framework/hosts/ipa.py (hunk 1 above)
Comment (pbrezina) [reply sssd_test_framework/hosts/ipa.py:201]: I don't know, I don't see any change here if I open the forced push diff.

**Comment by pbrezina** [comment]
I just rebased it, I'll push it once PR CI is finished.

**Pull request merged**